packages/browser/src/ClientAuthentication.ts

```diff
@@ -112,17 +113,21 @@ export default class ClientAuthentication {
   // if the expected information cannot be found.
   // Note that the ID token is not stored, which means the session information
   // cannot be validated at this point.
+  // FIXME: not actually part of the public interface:
   validateCurrentSession = async (
     currentSessionId: string
   ): Promise<(ISessionInfo & ISessionInternalInfo) | null> => {
     const sessionInfo = await this.sessionInfoManager.get(currentSessionId);
     if (
       sessionInfo === undefined ||
-      sessionInfo.clientAppId === undefined ||
+      // sessionInfo.clientAppId === undefined ||
       sessionInfo.issuer === undefined
     ) {
       return null;
     }
+
+    // FIXME: verify client is still valid
```

Comment:
This is actually taken care of by ClientRegistrar — if the client isn't valid, then a new one will be requested.

Valid Client: has clientId & clientExpiresAt == 0 or clientExpiresAt > now